scripts/author_stats.py

```diff
@@ -0,0 +1,120 @@
+"""Compute statistics about (non-Googler, non-bot) authors of commits."""
+
+from collections import defaultdict
+from datetime import datetime
+import csv
+import requests
+import sys
+import time
+
+# Constants
+GITHUB_API_URL = "https://api.github.com"
+REPO = "google/heir"  # replace with the desired repo
+# TOKEN = "your_github_token_here"  # Replace with your personal access token (if needed)
+
+stats = []
+
+# Define email patterns to exclude
+excluded_patterns = ["google.com", "j2kun", "kun.jeremy", "dependabot"]
+
+
+# Function to check if an email should be excluded
+def should_exclude(email):
+  return any(pattern in email for pattern in excluded_patterns)
+
+
+# Function to fetch all commits from a repository (with pagination)
+def fetch_commits(repo):
+  url = f"{GITHUB_API_URL}/repos/{repo}/commits"
+  # headers = {"Authorization": f"token {TOKEN}"}
+  params = {"per_page": 100}  # Fetch up to 100 commits per page
+  commits = []
+  page = 0
+
+  while url:
+    print(f"Fetching page {page} of commits...")
+    time.sleep(1)
+    response = requests.get(url, params=params)
+    response.raise_for_status()  # Raise an error if the request failed
+    data = response.json()
+    commits.extend(data)
+
+    # Check if there's another page of results
+    if "next" in response.links:
+      url = response.links["next"]["url"]
+      page += 1
+    else:
+      url = None
+
+  return commits
+
+
+def extract_commits(commits):
+  all_commits = []
+  for commit in commits:
+    commit_data = commit["commit"]
+    author_data = commit_data.get("author", {})
+    date_str = author_data.get("date")
+    email = author_data.get("email")
+    name = author_data.get("name")
+
+    if not email or should_exclude(email):
```

Comment:
answering my own question - private email addresses still have an email and that is username@noreply.github.com - so all is well for private email configs

guessing copybara and the integrate's HEIR team are also google.com ones?

Reply:
Correct, though I removed the generic google.com filter and added an LLVM integrate filter